### PR TITLE
[BugFix] Sorting nested keys

### PR DESCRIPTION
**Pull request by matteobettini**
## Description

This PR solves issue #775

### CODE REVIEW

File: torchrl/envs/vec_env.py [hunk 1]
```diff
@@ -39,6 +38,12 @@ def decorated_fun(self: _BatchedEnv, *args, **kwargs):
     return decorated_fun
 
 
+def _sort_keys(element):
+    if isinstance(element, tuple):
+        return "".join(element)
```
Comment (vmoens) [inline torchrl/envs/vec_env.py:43]: let's use a separator that is unlikely to be used in the tensordict
(for instance we may have a key `"nextobs"` and a key `("next", "obs")`)
I sometimes use `"_-|-_"` or similar, take your pick :)

**Review by vmoens — COMMENTED** [review]
LGTM, let's just make sure that we don't have keys that collide

File: torchrl/envs/vec_env.py (hunk 1 above)
Comment (matteobettini) [reply torchrl/envs/vec_env.py]: Thanks, this is very useful toknow. Fixed

**Comment by vmoens** [comment]
@matteobettini since the problem was occuring with Brax I updated the brax wrapper and tests to make sure that things were passing.
In general, it does not really make sense to execute brax in parallel envs (since it's vectorized), but it's a good way to spot bugs.

**Comment by matteobettini** [comment]
> @matteobettini since the problem was occuring with Brax I updated the brax wrapper and tests to make sure that 
things were passing.
> In general, it does not really make sense to execute brax in parallel envs (since it's vectorized), but it's a good way to spot bugs.

@vmoens 
I actually think it is important to use parallel envs with brax (and vmas).

This is because yes they perform batched simulation but parallelenv can furtherly distribute this to the various cpu cores.

This is also why rllib has the 2 parameters: num_workers and num_envs_per_worker

**Pull request merged**